azurelinuxagent/ga/exthandlers.py

```diff
@@ -273,6 +273,8 @@ def run(self):
                       message=detailed_msg)
             return
 
+        logger.periodic_info(logger.EVERY_HALF_HOUR, "[PERIODIC] WALinuxAgent HeartBeat")
```

Comment:
I'd do this in the main loop (UpdateHandler.run); since that code already logs this

>             logger.info(u"Agent {0} is running as the goal state agent", CURRENT_AGENT)

we might as well just log the same (periodically) to indicate the agent is still running

Reply:
This sounds like a better idea. I'll make the change.